module/surveySubmissions/helper.js

```diff
@@ -0,0 +1,421 @@
+   * @param {String} surveySubmissionId - survey submission id.
+   * @returns {JSON} consists of kafka message whether it is pushed for reporting
+   * or not.
+   */
+
+  static pushInCompleteSurveySubmissionForReporting(surveySubmissionId) {
+    return new Promise(async (resolve, reject) => {
+        try {
+
+            if (surveySubmissionId == "") {
+                throw new Error(messageConstants.apiResponses.SURVEY_SUBMISSION_ID_REQUIRED);
+            }
+
+            if(typeof surveySubmissionId == "string") {
+                surveySubmissionId = ObjectId(surveySubmissionId);
+            }
+
+            let surveySubmissionsDocument = await this.surveySubmissionDocuments
+            ({
+                _id: surveySubmissionId,
+                status: {$ne : "completed"}
+            })
+
+            if (!surveySubmissionsDocument.length) {
+                throw messageConstants.apiResponses.SUBMISSION_NOT_FOUND+"or"+messageConstants.apiResponses.SUBMISSION_STATUS_NOT_COMPLETE;
+            }
+        
+            const kafkaMessage = await kafkaClient.pushInCompleteSurveySubmissionToKafka(surveySubmissionsDocument[0]);
+
+            if(kafkaMessage.status != "success") {
+                let errorObject = {
+                    formData: {
+                        surveySubmissionId:surveySubmissionsDocument[0]._id.toString(),
+                        message:kafkaMessage.message
+                    }
+                };
+                slackClient.kafkaErrorAlert(errorObject);
+            }
+
+            return resolve(kafkaMessage);
+
+        } catch (error) {
+            return reject(error);
+        }
+    })
+}
+
+
+
+    /**
+    * Check if survey submission is allowed.
+    * @method
+    * @name isAllowed
+    * @param {String} submissionId - survey submissionId
+    * @param {String} evidenceId - evidence id
+    * @param {String} userId - logged in userId
+    * @returns {Json} - survey list.
+    */
+
+    static isAllowed(submissionId = "", evidenceId = "", userId = "") {
+        return new Promise(async (resolve, reject) => {
+            try {
+
+                if (submissionId == "") {
+                    throw new Error(messageConstants.apiResponses.SURVEY_SUBMISSION_ID_REQUIRED)
+                }
+
+                if (evidenceId == "") {
+                    throw new Error(messageConstants.apiResponses.EVIDENCE_ID_REQUIRED)
+                }
+                
+                if (userId == "") {
+                    throw new Error(messageConstants.apiResponses.USER_ID_REQUIRED_CHECK)
+                }
+                
+                let result = {
+                    allowed: true
+                };
+
+                let submissionDocument = await this.surveySubmissionDocuments
+                (
+                    { "_id": submissionId,
+                    },
+                    [
+                        "evidences." + evidenceId + ".isSubmitted",
```

Comment:
@deepakr123 - Can we instead of evidences use evidencesStatus. Please change in observation and assessments also.

Reply:
resolved